ee/codegen/src/generators/expression.ts

```diff
@@ -19,6 +15,17 @@ export declare namespace Expression {
   }
 }
 
+// This is to replace the usage of instanceof when checking for NodeInput
+// due to a circular dependency
+function isNodeInput(node: AstNode): boolean {
```

Comment:
```suggestion
NodeInput = AstNode & {
      nodeInputValuePointer: {
        nodeInputValuePointerData?: {
          rules?: Array<{ type: string }>;
        };
      };
    }

function isNodeInput(node: AstNode): node is NodeInput {
```

does this let us remove the cast below?

Reply:
Why yes it does